### DISCUSSION TITLE
Cannot build guest kernel for test suite anymore

### DISCUSSION THREAD

**Discussion by boisletterink** [strongswan/strongswan#1425] [General]
Hi,

I recently wanted to build the test suite again (using make-testing in the testing dir) but it failed on building the guest kernel. Turns out it can't download the ABI patch anymore.  Last time I tried (early September) it worked out of the box. I browsed around a bit on the server and found the patch elsewhere. Given that the script trying to download the patch (testing/scripts/build-guestkernel) is 6 years old, I suspect the directory containing the patches on the server has moved or a link/redirect has been removed.

It is easily worked around in the latest 5.9 branch by replacing the base URL in build-guestkernel on line 55:

```
-       url=http://download.strongswan.org/uml/$KERNELPATCH
+       url=http://download.strongswan.org/testing/$KERNELPATCH
```

However, fixing this in the source won't help when someone wants to build and test an older release. At the same time, not being able to update URLs means you have to keep cruft until the end of times...

If you want me to file this as a bug report, let me know.

Best regards,
Boi

**Comment by tobiasbrunner** [comment]
Thanks for the report. Fixed in master. The old URLs should also work again.
